src/main/java/Duke.java

```diff
@@ -111,22 +117,49 @@ void executeCommand(String command, String[] inputSplit) throws DukeException {
         } else if (command.equals("done")) {
             int taskInt = parseTaskInt(inputSplit[1]);
             System.out.println(lineWrap(markTask(taskInt)));
+            fileManager.updateFile(taskList, taskListPath);
         } else if (command.equals("todo") || command.equals("deadline") || command.equals("event")) {
             if (inputSplit.length == 1) {
                 throw new DukeException("☹ OOPS!!! The description of a " + command + " cannot be empty.");
             }
             Task task = createTask(inputSplit[1], command);
             System.out.println(lineWrap(addTask(task)));
+            fileManager.updateFile(taskList, taskListPath);
         } else if (command.equals("delete")) {
             int taskInt = parseTaskInt(inputSplit[1]);
             System.out.println(lineWrap(deleteTask(taskInt)));
+            fileManager.updateFile(taskList, taskListPath);
         } else {
             throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
         }
     }
 
+    private void loadFile(String filePath) {
+        try {
+            File f = new File(filePath); //hardCoded file directory
+            Scanner sc = new Scanner(f);
+            while (sc.hasNext()) {
+                String line = sc.nextLine();
+                String[] strArr = line.split(Pattern.quote(" | "));
```

Comment:
You can escape the line using `\\|` to avoid using Pattern.quote.

Reply:
Thanks, but I tried  `line.split("\\ | ");` and it does not seem to work due to the spacing. I could apply a `trim()` to all the elements in the array, but that would be an unnecessary addition of code.